src/main/webapp/DastFreeStyle.js

```diff
@@ -0,0 +1,1199 @@
+                fileType = grpcFileType;
+                elem = jq('#grpcFileId');
+                displayMessage = jq('#grpcUploadMessage');
+                break;
+            default :
+                throw new Exception("Illegal argument exception,File Type not valid");
+        }
+
+        if (file == null || fileType == null || elem == null) {
+            throw new Exception("Illegal argument exception,File Type not valid");
+        }
+
+        this.api.patchSetupManifestFile(this.releaseId, getAuthInfo(), file, fileType).then(res => {
+
+                //Todo: - check
+                console.log("File upload success " + res);
+                handleSpinner(uploadContainer, true);
+
+                if (res.fileId > 0) {
+                    elem.val(res.fileId);
+                    handleUploadStatusMessage(displayMessage, fileUploadSuccess, true);
+                } else {
+                    handleUploadStatusMessage(displayMessage, fileUploadFailed, false)
+                    throw new Exception("Illegal argument exception,FileId not valid");
+                }
+            }
+        ).catch((err) => {
+                handleSpinner(uploadContainer, true);
+                handleUploadStatusMessage(displayMessage, fileUploadFailed, false)
+                console.log('err' + err);
+            }
+        );
+    }
+
+    directoryAndSubdirectoriesScopeVisibility(isVisible) {
+        if (isVisible)
+            jq('#dast-standard-scan-scope').show();
+        else
+            jq('#dast-standard-scan-scope').hide();
+    }
+
+    onWorkflowDrivenHostChecked(event) {
+
+        let allowedHost = jq('#workflowMacroHosts').val();
+        if (event.target.checked) {
+            let hostToAdd = event.target.name; //name point to the host returned from FOD Patch API
+
+            if (allowedHost !== undefined || null) {
+                if (allowedHost.length > 0) {
+                    allowedHost = allowedHost + "," + hostToAdd;
+                } else
+                    allowedHost = hostToAdd;
+                jq('#workflowMacroHosts').val(allowedHost);
+            }
+        } else {
+            if (allowedHost !== undefined || null) {
+                let hosts = allowedHost.split(',');
+                hosts.forEach((entry) => {
+                    if (entry === event.target.name) {
+                        let index = hosts.lastIndexOf(entry);
+                        //remove
+                        if (index > -1) {
+                            hosts.splice(index, 1);
+                        }
+                    }
+                });
+                jq('#workflowMacroHosts').val(hosts.flat());
+            }
+        }
+    }
+
+    onApiTypeChanged() {
+
+        jq('.uploadMessage').text('');
+        this.onSourceChange('none');
+        let selectedApiTypeValue = jq('#apiTypeList').val();
+
+        if (selectedApiTypeValue === null || undefined) {
```

Comment:
Also fix this if statement like I commented above

Reply:
to find any remaining you can use this regex in Edit > Find > Find in Files

(undefined|null) +\|\| +(undefined|null)